Client/Frontend/Browser/TabManager.swift

```diff
@@ -379,9 +381,41 @@ extension TabManager {
 }
 
 extension TabManager : WKNavigationDelegate {
+    func webView(webView: WKWebView, didStartProvisionalNavigation navigation: WKNavigation!) {
+        if tabsNetworkActivity[webView] == nil {
+            tabsNetworkActivity[webView] = false
+        }
+        if tabsNetworkActivity[webView] == false {
+            tabsNetworkActivity[webView] = true
+            incrementNetworkActivityCount()
+        }
+    }
+
     func webView(webView: WKWebView, didFinishNavigation navigation: WKNavigation!) {
+        tabsNetworkActivity[webView] = false
+        decrementNetworkActivityCount()
+
         storeChanges()
     }
+
+    func webView(webView: WKWebView, didFailNavigation navigation: WKNavigation!, withError error: NSError) {
+        tabsNetworkActivity[webView] = false
+        decrementNetworkActivityCount()
+    }
+
+    func incrementNetworkActivityCount() {
+        networkActivityCount++
+        if networkActivityCount == 1 {
```

Comment:
I can change it to `> 0` but I think `== 1` is enough because we want to enable the spinner when the counter goes from 0 to 1.